Microsoft.AVS.Management/HcxUtils.ps1

```diff
@@ -106,3 +106,25 @@ function Get-HcxMetaData {
 
     throw "Unable To Retrieve HCX Manager Meta Data"
 }
+<#
+    .Synopsis
+    Get and return HCX Virtual machine
+    .Example
+    Get-HcxManagerVM
+#>
+function Get-HcxManagerVM {
+    Write-Host "Identifying HCX VM"
+    $HcxVm = $null
+    $VmsList = Get-VM
+
+    foreach ($Vm in $VmsList) {
+        if($Vm.Name.Contains("HCX-MGR")) {
+            $HcxVm = $Vm
+            break
+        }
+    }
+    if(-not $HcxVm) {
```

Comment:
I'd prefer that this check was done outside this function. For one thing throwing from here wouldn't stop the caller running, for another it's better to let the called decide what exactly they want to do about the absence of the VM - it makes this function easier to reason about and more useful.